swap_meet/vendor.py

```diff
@@ -1,2 +1,118 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        '''
+        Its parameter inventory is an empty list
+        '''
+        if not inventory:
+            inventory = []
+        self.inventory  = inventory
+
+    def add(self, add_item):
+        '''
+        It adds item to the inventory.
+        Argument - add_item(string): representing an item
+        Return: the item that was added 
+        '''
+        self.inventory.append(add_item)
+        return add_item
+
+    def remove(self, remove_item):
+        '''
+        It removes the matching item from the inventory.
+        Argument - remove_item(string): representing an item
+        Return: the item that was removed, if there is no matching item it returns False
+        '''
+        if remove_item in self.inventory:
+            self.inventory.remove(remove_item)
+            return remove_item
+        else:
+            return False
+
+    def get_by_category(self, category):
+        '''
+        It gets a list of the inventory and its category
+        Argument - category(string): representing a category
+        Return: a list of Items in the inventory with that category
+        '''
+        results = []
+        for item in self.inventory:
+            if category == item.category:
+                results.append(item)
+        return results
+
+    def swap_items(self, another_vendor, my_item, their_item):
+        '''
+        It swaps my item and my friend's item
+        Argument - 
+            another_vendor: the friend whom I(vendor but I will say I) am swapping my item with
+            my_item: the item I give to the friend
+            their_item: the item my friend gives to me
+        Return: if my inventory doesn't contain my_item or the friend's inventory doesn't contain their_item, it returns False, otherwise returns True
+        '''
+        if my_item not in self.inventory or their_item not in another_vendor.inventory:
+            return False
+        '''
+        Hi Audrey, I have two options here. 1. line 58 ~ 65, 2. line 68 ~ 71 
+        Which way would be a better way? :) 
+
+        another way of writing line  
+        my_inventory = self.inventory
+        friend_inventory = another_vendor.inventory
+
+        my_inventory.remove(my_item)
+        friend_inventory.append(my_item)
+        friend_inventory.remove(their_item)
+        my_inventory.append(their_item)
+        '''
+
+        self.inventory.remove(my_item)
+        another_vendor.inventory.append(my_item)
+        another_vendor.inventory.remove(their_item)
+        self.inventory.append(their_item)
```

Comment:
Great question! Both are very valid approaches.

I lean on Option 1 as slightly more readable. However, providing `self.inventory` and `another_vendor` a new name doesn't add much more value other than seeing one less dot notation, which isn't necessary as the code isn't too difficult to read in Option 2).

Reply:
Another valid approach is using the `remove` and `add` method for each vendor instance as we have already written the same logic in an earlier wave.